### PR TITLE
Add create pack and add sticker bot methods

### PR DESCRIPTION
**Pull request by YunusovSamat**
# Checklist

- [x] Linters and tests have passed
- [ ] I've written a changelog for PR change
- [x] I'll make a release when PR is approved

### CODE REVIEW

File: botx/client/stickers_api/add_sticker.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+)
+
+try:
+    from typing import Literal
+except ImportError:
+    from typing_extensions import Literal  # type: ignore  # noqa: WPS440
+
+
+class BotXAPIAddStickerRequestPayload(UnverifiedPayloadBaseModel):
+    sticker_pack_id: UUID
```
Comment (YunusovSamat) [inline botx/client/stickers_api/add_sticker.py:19]: note: added this path variable in payload

File: botx/client/stickers_api/stickers.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        stickers_order: Stickers positions.
+        stickers: Stickers data.
+
+    """
+
+    id: UUID
+    name: str
+    is_public: bool
+    stickers_order: List[UUID]
+    stickers: List[Sticker]
```
Comment (YunusovSamat) [inline botx/client/stickers_api/stickers.py:39]: note: I didn't add `insert_at`, `update_at`, `delete_at` and `preview`

File: botx/bot/bot.py [hunk 3]
```diff
@@ -687,6 +696,55 @@ async def send_internal_bot_notification(
...
+        * `name: UUID` - Sticker pack name.
+        """
+
+        payload = BotXAPICreateStickerPackRequestPayload.from_domain(name)
+        method = CreateStickerPackMethod(
+            bot_id,
+            self._httpx_client,
+            self._bot_accounts_storage,
+        )
+        sticker_pack = await method.execute(payload)
```
Comment (lxmnk) [inline botx/bot/bot.py:715]: `botx_api_sticker_pack`

File: botx/bot/bot.py [hunk 4]
```diff
@@ -687,6 +696,55 @@ async def send_internal_bot_notification(
...
+            sticker_pack_id,
+            emoji,
+            image_data,
+        )
+        method = AddStickerMethod(
+            bot_id,
+            self._httpx_client,
+            self._bot_accounts_storage,
+        )
+        sticker = await method.execute(payload)
```
Comment (lxmnk) [inline botx/bot/bot.py:744]: `botx_api_sticker`

File: botx/bot/bot.py [hunk 5]
```diff
@@ -687,6 +696,55 @@ async def send_internal_bot_notification(
...
         return botx_api_sync_id.to_domain()
 
+    # - Stickers API -
+    async def create_sticker_pack(self, bot_id: UUID, name: str) -> StickerPack:
+        """Create empty sticker pack.
+
+        **Arguments:**
+
+        * `bot_id: UUID` - Bot which should perform the request.
+        * `name: UUID` - Sticker pack name.
```
Comment (lxmnk) [inline botx/bot/bot.py:706]: `name: str`

File: botx/client/stickers_api/add_sticker.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        **AuthorizedBotXMethod.status_handlers,
+    }
+
+    async def execute(
+        self,
+        payload: BotXAPIAddStickerRequestPayload,
+    ) -> BotXAPIAddStickerResponsePayload:
+        json = payload.jsonable_dict()
+        sticker_pack_id = json.pop("sticker_pack_id")
+        path = f"/api/v3/botx/stickers/packs/{sticker_pack_id}/stickers"
```
Comment (lxmnk) [inline botx/client/stickers_api/add_sticker.py:65]: Pls add empty line before `path`.

File: setup.cfg [hunk 7]
```diff
@@ -47,7 +47,7 @@ allowed_domain_names = data, handler, result, content, file
 
 per-file-ignores =
     botx/bot/bot.py:WPS203
-    botx/__init__.py:WPS410,WPS412,F401,
+    botx/__init__.py:WPS410,WPS412,F401,WPS203
```
Comment (YunusovSamat) [inline setup.cfg:50]: :warning: add ignore: module with too many imported names

File: botx/__init__.py [hunk 8]
```diff
@@ -7,7 +7,7 @@
 from botx.bot.bot import Bot
 from botx.bot.exceptions import (
     AnswerDestinationLookupError,
-    BotShuttignDownError,
+    BotShuttingDownError,
```
Comment (lxmnk) [inline botx/__init__.py:10]: Pls move this typo fix to separate commit.

File: botx/bot/bot.py [hunk 9]
```diff
@@ -888,6 +897,57 @@ async def edit_message(
         )
         await method.execute(payload)
 
+    # - Stickers API -
+    async def create_sticker_pack(self, bot_id: UUID, name: str) -> StickerPack:
```
Comment (lxmnk) [inline botx/bot/bot.py:901]: Pls add make all arguments kw-only.

File: botx/bot/bot.py [hunk 10]
```diff
@@ -888,6 +897,57 @@ async def edit_message(
...
+        method = CreateStickerPackMethod(
+            bot_id,
+            self._httpx_client,
+            self._bot_accounts_storage,
+        )
+        botx_api_sticker_pack = await method.execute(payload)
+
+        return botx_api_sticker_pack.to_domain()
+
+    async def add_sticker(
```
Comment (lxmnk) [inline botx/bot/bot.py:920]: Pls make all arguments kw-only.

File: botx/bot/bot.py [hunk 11]
```diff
@@ -888,6 +897,57 @@ async def edit_message(
...
+        self,
+        bot_id: UUID,
+        sticker_pack_id: UUID,
+        emoji: str,
+        async_buffer: AsyncBufferReadable,
+        filename: str,
```
Comment (lxmnk) [inline botx/bot/bot.py:926]: Sticker doesn't need filename, does it?

File: botx/bot/bot.py [hunk 12]
```diff
@@ -888,6 +897,57 @@ async def edit_message(
...
+    ) -> Sticker:
+        """Add sticker in sticker pack.
+
+        * `bot_id: UUID` - Bot which should perform the request.
+        * `sticker_pack_id: UUID` - Sticker pack id to indicate where to add.
+        * `emoji: str` - Sticker emoji.
+        * `image: File` - Sticker image file.
```
Comment (lxmnk) [inline botx/bot/bot.py:933]: Update docstring.

File: botx/bot/bot.py [hunk 13]
```diff
@@ -952,9 +1012,9 @@ async def upload_file(
             self._httpx_client,
             self._bot_accounts_storage,
         )
-        api_async_file = await method.execute(payload, async_buffer, filename)
+        botx_api_async_file = await method.execute(payload, async_buffer, filename)
```
Comment (lxmnk) [inline botx/bot/bot.py:1015]: Pls move this typo fix to separate commit.

File: botx/client/stickers_api/add_sticker.py [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+        return cls(sticker_pack_id=sticker_pack_id, emoji=emoji, image=b64_content)
+
+
+class BotXAPIAddStickerResult(VerifiedPayloadBaseModel):
+    id: UUID
+    emoji: str
+    link: str
+    inserted_at: dt
+    updated_at: dt
+    deleted_at: Optional[dt]
```
Comment (lxmnk) [inline botx/client/stickers_api/add_sticker.py:57]: Don't validate fields that you don't use.

File: botx/client/stickers_api/add_sticker.py [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+class BotXAPIAddStickerResponsePayload(VerifiedPayloadBaseModel):
+    status: Literal["ok"]
+    result: BotXAPIAddStickerResult
+
+    def to_domain(self) -> Sticker:
+        return Sticker(
+            id=self.result.id,
+            emoji=self.result.emoji,
+            image_link=self.result.link,
```
Comment (lxmnk) [inline botx/client/stickers_api/add_sticker.py:68]: Can we use this link to download sticker?

File: botx/client/stickers_api/add_sticker.py [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+    status_handlers = {
+        **AuthorizedBotXMethod.status_handlers,
+        400: bad_request_error_handler,
+    }
+
+    async def execute(
+        self,
+        payload: BotXAPIAddStickerRequestPayload,
+    ) -> BotXAPIAddStickerResponsePayload:
+        json = payload.jsonable_dict()
```
Comment (lxmnk) [inline botx/client/stickers_api/add_sticker.py:98]: `json` -> `jsonable_dict`

File: botx/client/stickers_api/create_sticker_pack.py [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+class BotXAPICreateStickerPackRequestPayload(UnverifiedPayloadBaseModel):
+    name: str
+
+    @classmethod
+    def from_domain(cls, name: str) -> "BotXAPICreateStickerPackRequestPayload":
+        return cls(name=name)
+
+
+class BotXAPICreateStickerPackResult(VerifiedPayloadBaseModel):
```
Comment (lxmnk) [inline botx/client/stickers_api/create_sticker_pack.py:23]: Don't validate fields that you don't use.

File: scripts/format [hunk 18]
```diff
@@ -6,7 +6,16 @@ autoflake --recursive --in-place \
...
 black botx tests
+
+find botx -type f -name "*.py" | xargs add-trailing-comma --py36-plus --exit-zero-even-if-changed
+find tests -type f -name "*.py" | xargs add-trailing-comma --py36-plus --exit-zero-even-if-changed
+
+# This `black` is needed again in order to transfer parameters/arguments to new lines
+# after inserting commas.
+# The first `black` won't be able to transfer parameters/arguments to new lines because
+# there is no comma at the end of the line.
+# Inserting commas must be after the first `black`,so that there is one new line break,
```
Comment (lxmnk) [inline scripts/format:19]: `,so`

File: botx/models/attachments.py [hunk 19]
```diff
@@ -312,105 +312,105 @@ def decode_rfc2397(encoded_content: str) -> bytes:
...
-        ".arc": "application/x-freearc",
-        ".azw": "application/vnd.amazon.ebook",
-        ".bin": "application/octet-stream",
-        ".bz": "application/x-bzip",
-        ".bz2": "application/x-bzip2",
-        ".cda": "application/x-cdf",
-        ".csh": "application/x-csh",
-        ".doc": "application/msword",
-        ".docx": (
+        "7z": "application/x-7z-compressed",
```
Comment (lxmnk) [inline botx/models/attachments.py:315]: Pls move this fix to separate commit.

Comment (YunusovSamat) [reply botx/models/attachments.py:315]: This fix to a separate commit. The commit is "fix: files extensions"

File: botx/client/stickers_api/add_sticker.py (hunk 15 above)
Comment (YunusovSamat) [reply botx/client/stickers_api/add_sticker.py:68]: Yes, we can

**Review by Nimond — APPROVED** [review]
